src/learn/learn.cpp

```diff
@@ -971,15 +975,49 @@ namespace Learner
             local_loss_sum += loss;
             sum_norm += (double)abs(shallow_value);
 
-            // Determine if the teacher's move and the score of the shallow search match
-            const auto [value, pv] = Search::search(pos, 1);
-            if (pv.size() > 0 && (uint16_t)pv[0] == ps.move)
+            // Threat all moves with equal scores as first. This is up to move ordering.
+            if (has_depth1_move_agreement(pos, (Move)ps.move))
                 move_accord_count.fetch_add(1, std::memory_order_relaxed);
         }
 
         test_loss_sum += local_loss_sum;
     }
 
+    bool LearnerThink::has_depth1_move_agreement(Position& pos, Move pvmove)
+    {
+        // Determine if the depth 1 search pv matches the move from the dataset.
```

Comment:
Could we call Eval::evaluate() here to enable incremental computation of eval values?

Reply:
good point, done